frontend/app/routes/$lang+/_public+/apply+/$id+/adult/date-of-birth.tsx

```diff
@@ -142,23 +137,22 @@ export async function action({ context: { session }, params, request }: ActionFu
 
   saveApplyAdultState({ params, request, session, state: { dateOfBirth: parsedDataResult.data.dateOfBirth } });
 
-  const parseDateOfBirth = parse(parsedDataResult.data.dateOfBirth, 'yyyy-MM-dd', new Date());
-  const age = differenceInYears(new Date(), parseDateOfBirth);
-
-  if (age < 16) {
-    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/parent-or-guardian', params));
+  if (state.adultState.editMode) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/review-information', params));
   }
 
-  if (age === 16 || age === 17) {
-    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/living-independently', params));
-  }
+  const age = getAgeFromDateString(parsedDataResult.data.dateOfBirth);
 
-  if (age >= 18 && age < 65) {
+  if (age >= 18 && age <= 64) {
     return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/disability-tax-credit', params));
   }
 
-  if (state.adultState.editMode) {
-    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/review-information', params));
+  if (age >= 16 && age <= 17) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/living-independently', params));
+  }
+
+  if (age < 16) {
```

Comment:
**HUGE Nitpick:** elsewhere in this file the age check was explicitly modified to change the less-than to less-than-equal-to (was: `< 65`; changed to: `<= 64`). So why not stick to that convention and use `<= 15` here?

Reply:
I think we should create a function that returns the age group from age. 

``` ts
function getAgeGroup(age: number) {
  if(age >= 65) return "seniors" as const;
  if(age >= 18 && age < 65) return "adults"  as const;
  if(age >= 16 && age < 18) return "youth" as const;
  if(age > 0 && age < 16) return "children" as const;
  throw Error (`Invalid age [${age}]`);
}
```